src/source/lexstring.d

```diff
@@ -551,11 +548,23 @@ unittest {
 
 	// Check octal escape sequences.
 	checkLexString(`"\0\1\11\44\77\111\377"`, "\0\x01\x09\x24\x3f\x49\xff");
-	checkLexString(`"\1111\378"`, "\x491\x1f8");
+	checkLexString(`"\0\00\000"`, "\0\00\000");
+	checkLexString(
+		`"\01\001\02\002\03\003\04\004\05\005\06\006\07\007\08\008\09\009"`,
+		"\01\001\02\002\03\003\04\004\05\005\06\006\07\007\08\008\09\009");
+	checkLexString(`"\1\11\111"`, "\1\11\111");
+	checkLexString(`"\2\22\222"`, "\2\22\222");
+	checkLexString(`"\3\33\333"`, "\3\33\333");
+	checkLexString(`"\4\44"`, "\4\44");
+	checkLexString(`"\5\55"`, "\5\55");
+	checkLexString(`"\6\66"`, "\6\66");
+	checkLexString(`"\7\77"`, "\7\77");
```

Comment:
I would generally avoid to test octal literal against octal literals. If we were to use that lexer to compile this test, then the test would test nothing at all.

Some of these tests do not seem to test the behavior that is changed here. They should go in their own PR. If they are valuable, then we don't want to lose them is we come to revert that PR.